src/web3/createMulticallContractInstance.ts

```diff
@@ -0,0 +1,35 @@
+import MulticallAtrifact from "@uma/core/build/contracts/MulticallMock.json";
+import { ethers } from "ethers";
+
+// interface Network {
+//   [key: string]: {
+//     address: string;
+//     events: object;
+//     links: object;
+//     transactionHash: string;
+//   };
+// }
+
+// We need to query the data before a user has logged in
+// Because the provider doesn't have a default signer, we provide one with Void Signer, which creates a read only signer.
+// See: https://docs.ethers.io/v5/api/signer/#VoidSigner
+export default function createMulticallContractInstance(
+  signer: ethers.Signer
+  // networkId: string
+) {
+  // const artifact: Network = MulticallAtrifact.networks;
+  // const network = artifact[networkId];
+  // console.log("network", network, "artifact", artifact, MulticallAtrifact);
+
+  const contract = new ethers.Contract(
+    // network.address,
+    // This is stubbed in for now. We need to develop this formally for mainnet / kovan.
+    process.env.REACT_APP_CURRENT_ENV === "test"
+      ? "0xD6b61cAd80DBe2Fe26Fa672b8Ec2FdAf002CeBbC"
+      : "",
```

Comment:
nit: you can pull in pre-deployed addresses by network id by either importing the makerdao file that lists them via their package:
```
import multicallAddresses from "@makerdao/multicall/src/addresses";

const networkName = "kovan";
const multicallAddress = multicallAddresses[networkName];
```

OR what might be easier is just to copy the [mapping](https://github.com/makerdao/multicall.js/blob/master/src/addresses.json) into this file:
```
{
  "mainnet": {
    "multicall": "0xeefba1e63905ef1d7acba5a8513c70307c1ce441",
    "rpcUrl": "https://mainnet.infura.io"
  },
  "kovan": {
    "multicall": "0x2cc8688c5f75e365aaeeb4ea8d6a480405a48d2a",
    "rpcUrl": "https://kovan.infura.io"
  },
  "rinkeby": {
    "multicall": "0x42ad527de7d4e9d9d011ac45b31d8551f8fe9821",
    "rpcUrl": "https://rinkeby.infura.io"
  },
  "goerli": {
    "multicall": "0x77dca2c955b15e9de4dbbcf1246b4b85b651e50e",
    "rpcUrl": "https://rpc.slock.it/goerli"
  },
  "xdai": {
    "multicall": "0xb5b692a88bdfc81ca69dcb1d924f59f0413a602a",
    "rpcUrl": "https://dai.poa.network"
  }
}
```

You could remove the RPC urls and change the names to network IDs to make it easier to look up at runtime.

Reply:
Good to know. If there are already kovan / mainnet deployments out there, we just need to worry about the locally deployed ones.